src/Analyser.php

```diff
@@ -473,20 +515,40 @@ private function initExistingSymbols(): void
             'Composer\\Autoload\\ClassLoader' => true,
         ];
 
-        /** @var string $constantName */
-        foreach (get_defined_constants() as $constantName => $constantValue) {
-            $this->ignoredSymbols[$constantName] = true;
+        /** @var array<string, mixed> $constants */
+        foreach (get_defined_constants(true) as $constantExtension => $constants) {
+            foreach ($constants as $constantName => $_) {
+                if ($constantExtension === 'user') {
+                    $this->ignoredSymbols[$constantName] = true;
+                } else {
+                    $extensionName = 'ext-' . $constantExtension;
+
+                    if (in_array($extensionName, self::CORE_EXTENSIONS, true)) {
+                        $this->ignoredSymbols[$constantName] = true;
+                    } else {
+                        $this->extensionSymbols[SymbolKind::CONSTANT][$constantName] = $extensionName;
+                    }
+                }
+            }
         }
 
         foreach (get_defined_functions() as $functionNames) {
             foreach ($functionNames as $functionName) {
                 $reflectionFunction = new ReflectionFunction($functionName);
                 $functionFilePath = $reflectionFunction->getFileName();
 
-                if ($reflectionFunction->getExtension() === null && is_string($functionFilePath)) {
-                    $this->definedFunctions[$functionName] = Path::normalize($functionFilePath);
+                if ($reflectionFunction->getExtension() === null) {
+                    if (is_string($functionFilePath)) {
+                        $this->definedFunctions[$functionName] = Path::normalize($functionFilePath);
+                    }
                 } else {
-                    $this->ignoredSymbols[$functionName] = true;
+                    $extensionName = 'ext-' . $reflectionFunction->getExtension()->name;
```

Comment:
I think the extension name should be always normalized to lowercase

Reply:
Yes, and composer also normalizes spaces to dashes: https://github.com/maglnet/ComposerRequireChecker/issues/99